USFMConverter/UI/Pages/ProjectDetailScreen.axaml.cs

```diff
@@ -0,0 +1,170 @@
+using System;
+using Avalonia.Controls;
+using Avalonia.Markup.Xaml;
+using Avalonia.Interactivity;
+using USFMConverter.Core;
+using USFMConverter.UI.Pages.PartialView;
+
+namespace USFMConverter.UI.Pages
+{
+    public partial class ProjectDetailScreen : UserControl
+    {
+        private OptionView optionView;
+        private FileView fileView;
+        private ProgressBar progressBar;
+        private Button openOptionBtn;
+        private StackPanel backgroundOverlay;
+        private UserControl progressDialog;
+        private Success successDialog;
+        private Error errorDialog;
+        private TextBlock errorMessageText;
+
+        public event EventHandler<RoutedEventArgs> ShowBackgroundOverlay
+        {
+            add
+            {
+                AddHandler(ShowBackgroundOverlayEvent, value);
+            }
+            remove
+            {
+                RemoveHandler(ShowBackgroundOverlayEvent, value);
+            }
+        }
+
+        public event EventHandler<RoutedEventArgs> HideBackgroundOverlay
+        {
+            add
+            {
+                AddHandler(HideBackgroundOverlayEvent, value);
+            }
+            remove
+            {
+                RemoveHandler(HideBackgroundOverlayEvent, value);
+            }
+        }
+
+        public event EventHandler<RoutedEventArgs> StartNewProject
+        {
+            add
+            {
+                AddHandler(StartNewProjectEvent, value);
+            }
+            remove
+            {
+                RemoveHandler(StartNewProjectEvent, value);
+            }
+        }
+
+        public static readonly RoutedEvent<RoutedEventArgs> ShowBackgroundOverlayEvent =
```

Comment:
This is how it should be done with events not reaching into other controls by getting an element by its name

Reply:
yay